src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -72,7 +72,10 @@ public function __construct(array $defaultOptions = [], int $maxHostConnections
             curl_multi_setopt($this->multi->handle, CURLMOPT_PIPELINING, CURLPIPE_MULTIPLEX);
         }
         if (\defined('CURLMOPT_MAX_HOST_CONNECTIONS')) {
-            curl_multi_setopt($this->multi->handle, CURLMOPT_MAX_HOST_CONNECTIONS, 0 < $maxHostConnections ? $maxHostConnections : PHP_INT_MAX);
+            $maxHostConnections = curl_multi_setopt($this->multi->handle, CURLMOPT_MAX_HOST_CONNECTIONS, 0 < $maxHostConnections ? $maxHostConnections : PHP_INT_MAX) ? 0 : $maxHostConnections;
```

Comment:
PHP docs says that `curl_multi_setopt()` returns a `bool`.

Reply:
Not sure to get what you mean, did you miss the ternary operation maybe?

Reply:
Of course I did :)